2016-zone-b/readme.md

```diff
@@ -0,0 +1,304 @@
+# 2016 Zone B Answers
+
+| **DISCLAIMER**  |
+| :---: |
+| This repository of documents and source codes were created with the intention of helping fellow students. No infringement of copyrights or legal rights intended. |
+| Please only take these answers as a *reference* for your research purposes. |
+
+---
+
+## Question 1
+
+* a) i) int
+* a) ii) char
+* a) iii) short
+* a) iv) double
+
+---
+
+* b) i) int,Long
+* b) ii) double
+* b) iii) boolean
+* b) iv) char
+
+---
+
+* c)
+```java
+public class Sample{
+  public static void main(String[] args){
+    
+  }
+}
+```
+
+---
+
+* d)
+```java
+public class Count{
+  public static void main(String[] args){
+    System.out.println(args.length);
+  }
+}
+```
+
+---
+
+* e)
+```java
+public class ToDoList{
+  private String item;
+  private boolean done;
+  
+  public ToDoList(String item, boolean done){
+    this.item = item;
+    this.done = done;
+  }
+  
+  public boolean isDone(){
+    return done;
+  }
+  
+  public String toString(){
+    String s = ("The to do list item: "+item+"\n");
+    
+    if(isDone()) s = s + "has been done";
+    else s = s + "has not been done";
+    
+    return s;
+  }
+  
+  public static void main(String[] args){
+    ToDoList toDoList = new ToDoList("take cat to vet",false);
+    System.out.println(toDoList);
+  }
+}
+```
+
+---
+
+## Question 2
+
+* a) i) yes
+* a) ii) yes
+* a) iii) no
+* a) iv) yes
+* a) v) yes
+* a) vi) yes
+* a) vii) yes
+
+---
+
+* b) (A)
+
+---
+
+* c) i) 12
+* c) ii) 6
+* c) iii) Infinite loop
+* c) iv) 22
+* c) v) Infinite loop
+* c) vi) 0
+* c) vii) 10
+* c) viii) 10
+
+---
+
+* d)
+```java
+static int shortestLength(String [] arr){
+int short = 0;
+if(arr.length > 0) short = a[0];
+  for(int i=1; i<arr.length; i++){
+    if(short > arr[i]){
+      short = arr[i];
+    }
+  }
+return short;
+}
+```
+---
+
+## Question 3
+
+* a) i) Cannot assign String to int.
+* a) ii) Cannot declare a variable again.
+* a) iii) Class D has an error, because String z must be static in order to be used in static method.
+
+---
+
+* b) i)
+```java
+if(t 1 > t2 && b) t1 = t2+1;
+System.out.println(t1);
+```
+* b) ii)
+```java
+if( b != true) { y = 7 + 10; } 

```

Comment:
Did the question specifically state that `y` must be assigned to the expression `7 + 10`? Could you directly assign `y = 17;`?